include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -884,7 +884,7 @@ struct __gen_mask
         return __pred((__rng_transform(std::forward<_InRng>(__in_rng)))[__id]);
     }
     _Predicate __pred;
-    _RangeTransform __rng_transform;
+    _RangeTransform __rng_transform{};
```

Comment:
I guess we have got problem with lambda before C++20...

Reply:
That's right.

I've changed the approach to pass `__no_op` explicitly. Also, I removed a default template parameter to make sure it `__no_op` is explicitly passed in the future.